### PR TITLE
Merge heyting-algebras

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

File: src/Algebra/Lattice/Boolean.hs [hunk 1]
```diff
@@ -0,0 +1,156 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+module Algebra.Lattice.Boolean
+  ( BooleanAlgebra
+  , (==>)
+  , not
+  , iff
```
Comment (phadej) [inline src/Algebra/Lattice/Boolean.hs]: we could have `<=>` alias for this. And `implies` alias for `==>`

**Comment by coot** [comment]
in `heyting-algebras` I had this cabal flag `EXPORT_PROPERTIES` which if unset will not export quickcheck properties (then there is one transitive dependency less, i.e. quickcheck).  We should either:
* remove it
* add a build that tests when it is unset
What do you prefer?

File: src/Algebra/Lattice/Heyting.hs [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+--
+-- Some other useful properties of Heyting algebras:
+-- 
+-- prop> (a ⇒ b) ∧ a ≤ b
+-- prop> b ≤ a ⇒ a ∧ b
+-- prop> a ≤ b  iff a ⇒ b = ⊤
+-- prop> b ≤ b' then a ⇒ b ≤ a ⇒ b'
+-- prop> a'≤ a  then a' ⇒ b ≤ a ⇒ b
+-- prop> not (a ∧ b) = not (a ∨ b)
+-- prop> not (a ∨ b) = not a ∧ not b
```
Comment (phadej) [inline src/Algebra/Lattice/Heyting.hs]: note to self: https://en.wikipedia.org/wiki/Heyting_algebra#The_De_Morgan_laws_in_a_Heyting_algebra

Add an counterexample.

**Comment by coot** [comment]
I think we can close this one, since `lattice` package include Heyting algebras by now.

**Pull request closed without merging**